packages/zundo/src/types.ts

```diff
@@ -26,9 +27,8 @@ export interface ZundoOptions<TState, PartialTState = TState> {
   limit?: number;
   equality?: (currentState: TState, pastState: TState) => boolean;
   onSave?: onSave<TState>;
-  handleSet?: (
-    handleSet: StoreApi<TState>['setState'],
-  ) => StoreApi<TState>['setState'];
+  handleSet?: (handleSet: StoreApi<TState>['setState']) => StoreApi<TState>['setState'];
+  storeWrap?: (initializer: StateCreator<TemporalStateWithInternals<TState>, [], []>) => StateCreator<TemporalStateWithInternals<TState>, [], []>
```

Comment:
What other names did you think of for storeWrap? I'm wondering if there is a better one we could use. 

Perhaps, temporalMiddleware or simply middleware? Maybe storeMiddleware or wrapTemporalStore?

I'm not sure if there are other examples is in the zustand ecosystem that use this pattern we could copy. 

What do you think?

Reply:
storeWrap was more of a WIP TBD name, either `middlewares` or `middlewareInjection` would be much more suitable

alternatively, this could be called `storeInitializerInjection` since the purpose if it is to add some custom functionality e.g. middlewares to a store initializator which is the zustand name for the function we pass to `create` method

Reply:
Okay, I prefer the name `wrapTemporal` or `wrapTemporalStore`. 

What do you think?

Once these small comments are addressed, we can merge this in, add docs, and release a new version! Thank you for all the great help.

Reply:
Happy to be working on this, amazing opportunity to both learn more about middlewares as well as to do some good for the people using this much like i do

`wrapTemporalStore` sounds good to me